### PR TITLE
[Feature] Support the test of PrDiMP

### PR DESCRIPTION
**Pull request by JingweiZhang12**

### CODE REVIEW

File: mmtrack/models/sot/__init__.py [hunk 1]
```diff
@@ -1,5 +1,6 @@
 # Copyright (c) OpenMMLab. All rights reserved.
+from .prdimp import Prdimp
 from .siamrpn import SiamRPN
 from .stark import Stark
 
-__all__ = ['SiamRPN', 'Stark']
+__all__ = ['SiamRPN', 'Stark', 'Prdimp']
```
Comment (GT9505) [inline mmtrack/models/sot/__init__.py:6]: Prdimp or PrDIMP?

File: mmtrack/models/task_modules/__init__.py [hunk 2]
```diff
@@ -1,4 +1,5 @@
 # Copyright (c) OpenMMLab. All rights reserved.
 from .anchor import *  # noqa: F401,F403
+from .filter import *  # noqa: F401,F403
```
Comment (GT9505) [inline mmtrack/models/task_modules/__init__.py:3]: We have `filter` both in `mmtrack/models/filter` and `mmtrack/models/task_modules/filter`.
Better to rename one of them.
Maybe `filter_modules` and `filter`?

File: mmtrack/models/track_heads/__init__.py [hunk 3]
```diff
@@ -1,4 +1,6 @@
 # Copyright (c) OpenMMLab. All rights reserved.
+from .iounet_head import IouNetHead
+from .prdimp_cls_head import PrdimpClsHead
```
Comment (GT9505) [inline mmtrack/models/track_heads/__init__.py:3]: same here. `PrdimpClsHead` or `PrDIMPClsHead`?

File: mmtrack/models/task_modules/filter/filter.py [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+            activation.reshape(-1, 1, *activation.shape[-2:]),
+            padding=transpose_pad,
+            groups=num_img_per_seq * batch_size)
+
+        # The shape of returns is (bs, c, filter_h, fiter_w)
+        if num_img_per_seq == 1:
+            return filter_grad.permute(1, 0, 2, 3)
+        return filter_grad.view(-1, num_img_per_seq, batch_size,
+                                *filter_grad.shape[-2:]).sum(dim=1).permute(
+                                    1, 0, 2, 3)
```
Comment (GT9505) [inline mmtrack/models/task_modules/filter/filter.py:85]: better to simplify the code by decomposing into multiple lines

File: mmtrack/models/task_modules/filter/filter.py [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+        filter_grad = F.conv2d(
+            activation.reshape(1, -1, *activation.shape[-2:]),
+            feat.reshape(-1, 1, *feat.shape[-2:]),
+            padding=transpose_pad,
+            groups=batch_size)
+
+        # The shape of returns is (1, c, filter_h, fiter_w)
+        return filter_grad.view(batch_size, 1, -1,
+                                *filter_grad.shape[-2:]).sum(dim=0).flip(
+                                    (2, 3))
```
Comment (GT9505) [inline mmtrack/models/task_modules/filter/filter.py:99]: better to simplify the code by decomposing into multiple lines

File: mmtrack/utils/misc.py [hunk 6]
```diff
@@ -102,3 +103,25 @@ def convert_data_sample_type(
                 ref_data_samples[i].pop(key)
 
     return ref_data_samples, ref_metainfos
+
+
+def max2d(input: Tensor) -> Tuple[Tensor, Tensor]:
```
Comment (GT9505) [inline mmtrack/utils/misc.py:108]: maybe `tensor_max_last2d`?

File: mmtrack/models/task_modules/__init__.py (hunk 2 above)
Comment (JingweiZhang12) [reply mmtrack/models/task_modules/__init__.py:3]: But we also have this convention: `mmtrack/models/motion` and `mmtrack/motion`

File: mmtrack/models/sot/__init__.py (hunk 1 above)
Comment (JingweiZhang12) [reply mmtrack/models/sot/__init__.py:6]: We rename all names containing `prdimp` as `Prdimp`

File: mmtrack/utils/misc.py (hunk 6 above)
Comment (JingweiZhang12) [reply mmtrack/utils/misc.py:108]: `max_last2d` may be better?

File: demo/demo_mot_vis.py [hunk 7]
```diff
@@ -103,7 +103,7 @@ def main(args):
         # show the results
         visualizer.add_datasample(
             'mot',
-            img,
+            img[..., ::-1],
```
[the remark this answers is not in the document]
Comment (JingweiZhang12) [reply demo/demo_mot_vis.py:106]: Yes

**Review by GT9505 — COMMENTED** [review]
we can merge this pr after mmcv support prroipooling

**Review by GT9505 — APPROVED** [review]

**Pull request merged**